### ISSUE TITLE
Incorrect fill_betweenx interpolation

### ISSUE THREAD

**Issue by dopplershift** [matplotlib/matplotlib#9235]
This sample:
```python
import matplotlib.pyplot as plt
import numpy as np

p = np.array([724.3, 700, 655])
t = np.array([9.4, 7, 2.2])
prof = np.array([7.9, 6.6, 3.8])

fig = plt.figure(figsize=(9, 9))
ax = fig.add_subplot(1, 1, 1)

ax.plot(t, p, 'tab:red')
ax.plot(prof, p, 'k')

ax.fill_betweenx(p, t, prof, where=prof<t,
                            facecolor='blue', interpolate=True, alpha=0.4)
ax.fill_betweenx(p, t, prof, where=prof>t,
                            facecolor='red', interpolate=True, alpha=0.4)

ax.set_xlim(0, 30)
ax.set_ylim(800, 600)
```
is producing incorrect results on 2.1rc:

![fill_between](https://user-images.githubusercontent.com/221526/30874047-7819748a-a2ac-11e7-8246-c8a7d1b41be9.png)

**Comment by dopplershift** [comment]
Ah, it works if I reverse all of the data arrays.

**Comment by tacaswell** [comment]
Closed by #9236
